File: leetcode/每日一题/2023/05/13/solution.java
class Solution {
    public int findMaxK(int[] nums) {
        List<Integer> set = new ArrayList<>();
        Arrays.sort(nums);
        for(int i=0;i<nums.length;++i){
            set.add(nums[i]);
        }
        int res = -1;
        for(int i = 0;i<set.size();++i){
            if(set.contains(-set.get(i))){
                res = Math.max(Math.abs(set.get(i)),res);
            }
        }
        return res;
    }
}

File: leetcode/每日一题/2023/05/13/solution.go
func findMaxK(nums []int) int {
	set := make(map[int]bool)
	n := len(nums)
	for i := 0;i<n;i++{
		set[nums[i]] = true
	}

	res := -1
	for i := 0;i<n;i++{
		if nums[i] > 0 && set[-nums[i]]{
			res = max(res,nums[i])
		}
	}
	return res
}

func max(a int,b int) int{
    if a > b{
        return a
    }else{
        return b
    }
}